include/zephyr/drivers/timer/system_timer.h

```diff
@@ -160,6 +160,16 @@ uint32_t sys_clock_cycle_get_32(void);
  */
 uint64_t sys_clock_cycle_get_64(void);
 
+
+/**
+ * @brief Initialize system clock driver
+ * @note This function is called by the kernel during system initialization.
+ *       It should not be called by application code.
+ * @return 0 on success, negative errno code on fail
+ */
+int init_sys_clock_driver(void);
```

Comment:
this prototype should likely not be exposed in the docs, otherwise it may seem like a public API (even if stated it's not like that).

Ideally we should have an internal level of headers, where APIs to be consumed by Zephyr could be exposed to the Kernel/arch/soc code, but not to the application.

Reply:
fully agree.

Reply:
Also, should we prefix these with `z_`? To avoid any `init_` name collisions and to make it clearer that these aren't supposed to be called by the application.

Reply:
yeah, I am working my way through all of this and already thinking about a unified prefix for all those initialization calls. we now have a mix of `z_` , `init_`, `..._init()` etc.
This PR is just trying to show the potential of how we could fix mis-use of SYS_INIT and reduce the boot dependency problem to something manageable.